### PR TITLE
StorageClass - add support for resource creation

### PR DESCRIPTION
**Pull request by dalia-frank**
##### Short description: 
StorageClass - add support for resource creation

##### More details: 
Add missing args to create functional storage class

https://kubernetes.io/docs/concepts/storage/storage-classes/

##### What this PR does / why we need it: 
In order to create a storage class object

### CODE REVIEW

File: ocp_resources/storage_class.py [hunk 1]
```diff
@@ -45,3 +45,57 @@ class Annotations:
...
             f"{Resource.ApiGroup.STORAGECLASS_KUBERNETES_IO}/is-default-class"
         )
+
+    class ReclaimPolicy:
+        DELETE = "Delete"
+        RETAIN = "Retain"
+
+    def __init__(
+        self,
+        name=None,
```
Comment (myakove) [inline ocp_resources/storage_class.py:55]: Please leave here only what this specific resource needs (like reclaim_policy, provisioner , etc)
all other will be passed with kwargs

File: ocp_resources/storage_class.py [hunk 2]
```diff
@@ -45,3 +45,57 @@ class Annotations:
...
+        self.provisioner = provisioner
+        self.reclaim_policy = reclaim_policy
+        self.volume_binding_mode = volume_binding_mode
+        self.allow_volume_expansion = allow_volume_expansion
+        self.parameters = parameters
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.update({"provisioner": self.provisioner})
```
Comment (myakove) [inline ocp_resources/storage_class.py:93]: is `self.provisioner` mandatory? 
if so please raise if it isn't passed

File: ocp_resources/storage_class.py [hunk 3]
```diff
@@ -45,3 +45,57 @@ class Annotations:
...
+            allow_volume_expansion (bool): True for allowing the volume expansion
+            parameters (list): Describe volumes belonging to the storage class.
+        """
+        super().__init__(
+            name=name,
+            client=client,
+            teardown=teardown,
+            **kwargs,
+        )
+        self.provisioner = provisioner
```
Comment (myakove) [inline ocp_resources/storage_class.py:84]: Do you mind add also
`mountOptions` and `allowedTopologies`

**Review by myakove — CHANGES_REQUESTED** [review]
Please edit the PR title to be more descriptive

File: ocp_resources/storage_class.py (hunk 1 above)
Comment (dalia-frank) [reply ocp_resources/storage_class.py]: ack

File: ocp_resources/storage_class.py (hunk 2 above)
Comment (dalia-frank) [reply ocp_resources/storage_class.py:96]: ack

File: ocp_resources/storage_class.py (hunk 3 above)
Comment (dalia-frank) [reply ocp_resources/storage_class.py:83]: done

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Review by rnetser — CHANGES_REQUESTED** [review]
1. please add a link to the api
2. please updtae the pr title - the title goes into the release notes and should describe whart this pr is about

**Review by rnetser — CHANGES_REQUESTED** [review]
My comment about PR title was not addressed; please rephrase

**Comment by dalia-frank** [comment]
> My comment about PR title was not addressed; please rephrase

I updated it in the wrong place, fixed

**Review by rnetser — APPROVED** [review]

**Pull request merged**